generator/views/datagroups.py

```diff
@@ -56,121 +55,119 @@ def __lt__(self, other) -> bool:
-    return [
-        Datagroup(
-            name=f"{table_id}_last_updated",
-            label=f"{table_metadata.get('friendlyName', table_id) or table_id} Last Updated",
-            description=f"Updates when {full_table_id} is modified.",
-            sql_trigger=SQL_TRIGGER_TEMPLATE.format(
-                project_id=project_id,
-                dataset_id=dataset_id,
-                table_id=table_id,
-            ),
-            # Note: Ideally we'd use the table's ETL schedule as the `maximum_cache_age` but we don't have schedule
-            # metadata here.
-        ),
-        # create a datagroup associated to a view which will be used for caching
-        Datagroup(
-            name=f"{view.name}_last_updated",
-            label=f"{view.name} Last Updated",
-            description=f"Updates for {view.name} when {full_table_id} is modified.",
-            sql_trigger=SQL_TRIGGER_TEMPLATE.format(
-                project_id=project_id,
-                dataset_id=dataset_id,
-                table_id=table_id,
-            ),
+    if len(tables) == 0:
+        return None
+
+    datagroup_tables = []
+    for table in tables:
+        dataset_id = table[1]
+        table_id = table[2]
+
+        datagroup_tables.append(
+            f"(table_schema = '{dataset_id}' AND table_name = '{table_id}')"
+        )
+
+    # create a datagroup associated to a view which will be used for caching
+    return Datagroup(
+        name=f"{view.name}_last_updated",
+        label=f"{view.name} Last Updated",
+        description=f"Updates for {view.name} when referenced tables are modified.",
+        sql_trigger=SQL_TRIGGER_TEMPLATE.format(
+            project_id=project_id, tables=" OR ".join(datagroup_tables)
         ),
-    ]
+    )
 
 
 def _get_datagroup_from_bigquery_view(
     project_id,
     dataset_id,
     table_id,
-    table_metadata: Dict[str, Any],
     dataset_view_map: BQViewReferenceMap,
-    dryrun,
     view: View,
-) -> List[Datagroup]:
+) -> Optional[Datagroup]:
     # Dataset view map only contains references for shared-prod views.
     full_table_id = f"{project_id}.{dataset_id}.{table_id}"
     if project_id not in ("moz-fx-data-shared-prod", "mozdata"):
         logging.debug(
             f"Unable to get sources for non shared-prod/mozdata view: {full_table_id} in generated-sql."
         )
-        return []
+        return None
+
+    view_references = _get_referenced_tables(
+        project_id, dataset_id, table_id, dataset_view_map
+    )
+
+    if not view_references or len(view_references) == 0:
+        # Some views might not reference a source table
+        logging.debug(f"Unable to find a source for {full_table_id} in generated-sql.")
+        return None
+
+    return _get_datagroup_from_bigquery_tables(project_id, view_references, view)
+
+
+def _get_referenced_tables(
+    project_id,
+    dataset_id,
+    table_id,
+    dataset_view_map: BQViewReferenceMap,
+    seen: List[List[str]] = [],
```

Comment:
Since the default value is mutable, the changes to the list will persist across calls which I don't think is desired.  If it is then I think it's confusing behaviour.  It would be better to make the default value `None` or not have a default value and have the caller pass in an empty list